### ISSUE TITLE
Poor error message when missing required encoding configuration for `aws_kinesis_firehose`

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#12473]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Not providing encoding configuration for the `aws_kinesis_firehose` sink now returns:

```
2022-04-28T22:55:54.593972Z ERROR vector::cli: Configuration error. error=data did not match any variant of untagged enum EncodingConfigAdapter for key `sinks.sink0`
```

I believe as of https://github.com/vectordotdev/vector/pull/12176

We should add some tests around this when fixing it. I'm guessing other sinks that have been converted might be affected by this too.

### Version

vector 0.22.0 (aarch64-apple-darwin debug=full)

**Comment by pablosichert** [comment]
Providing `encoding` has always been mandatory, here for `0.21.1`:

https://github.com/vectordotdev/vector/blob/18787c02047fb516ef05323cb3935d554d0bcf00/src/sinks/aws_kinesis_firehose/config.rs#L54

https://github.com/vectordotdev/vector/blob/18787c02047fb516ef05323cb3935d554d0bcf00/src/sinks/util/encoding/config.rs#L21-L25

https://github.com/vectordotdev/vector/blob/18787c02047fb516ef05323cb3935d554d0bcf00/src/sinks/aws_kinesis_firehose/config.rs#L98-L105

So as far as I'm aware this is not a bug.

I agree that the error message you are seeing is poor, though. This is a known issue and artifact of #12162

**Comment by pablosichert** [comment]
If you try that in `0.21.1` when omitting the `encoding` key you get:

> ERROR vector::cli: Configuration error. error=missing field `encoding` for key `sinks.sink`

and when omitting the `encoding.codec` key:

> ERROR vector::cli: Configuration error. error=missing field `codec` for key `sinks.sink`

**Comment by jszwedko** [comment]
Also, apologies, I did initially think it was the missing default that was the issue, but I see that behavior didn't change, just the error message you receive.

**Comment by pablosichert** [comment]
No worries!

I agree though that this is bad enough that we want to get it fixed before `v0.22.0`. And try to get the `#[serde(deny_unknown_fields)]` back in again on the sinks that integrate with `Encoder`
